sharding/handler/smc_handler.py

```diff
@@ -34,25 +35,33 @@
 smc_json = get_smc_json()
 
 
-class SMCHandler(Contract):
+class SMC(Contract):
 
-    logger = logging.getLogger("evm.chain.sharding.SMCHandler")
+    logger = logging.getLogger("evm.chain.sharding.SMC")
     abi = smc_json["abi"]
     bytecode = decode_hex(smc_json["bytecode"])
 
-    private_key = None  # type: datatypes.PrivateKey
-    sender_address = None  # type: Address
+    default_priv_key = None  # type: datatypes.PrivateKey
+    default_sender_address = None  # type: Address
     config = None  # type: Dict[str, Any]
 
+    _estimate_gas_dict = dict()  # type: Dict[str, Any]
+
     def __init__(self,
                  *args: Any,
-                 private_key: datatypes.PrivateKey,
+                 default_priv_key: datatypes.PrivateKey,
                  config: Dict[str, Any],
                  **kwargs: Any) -> None:
-        self.private_key = private_key
-        self.sender_address = self.private_key.public_key.to_canonical_address()
+        self.default_priv_key = default_priv_key
+        self.default_sender_address = self.default_priv_key.public_key.to_canonical_address()
         self.config = config
 
+        # Extract estimate gas from abi
+        for entry in self.abi:
+            if entry['type'] == 'function':
+                if not entry['constant']:
+                    self._estimate_gas_dict[entry['name']] = entry['gas']
```

Comment:
This can be simplified:
```
self._estimate_gas_dict = {
    entry['name']: entry['gas']
    for entry in self.abi
    if entry['type'] == 'function' and not entry['constant']
}
```